app/controllers/items_controller.rb

```diff
@@ -91,4 +93,12 @@ def scrape_with_url(raw_url)
   def item_already_exists?
     @item.errors[:item_url][0] == "has already been taken"
   end
+
+  private
+
+  def authorize_user
+    if !user_signed_in?
```

Comment:
Would it be more useful to tell the user that they need to sign in first, using a `flash[:error]` message and redirecting them to the sign in page?